tools/editor/editor_node.cpp

```diff
@@ -6267,11 +6294,71 @@ EditorNode::EditorNode() {
 
 EditorNode::~EditorNode() {
 
-
+        
 	memdelete( EditorHelp::get_doc_data() );
 	memdelete(editor_selection);
+        memdelete(editor_plugins_over);
 	memdelete(file_server);
 	EditorSettings::destroy();
 }
+/*
+ * EDITOR PLUGIN LIST
+ */
+
+
+void EditorPluginList::make_visible(bool p_visible) {
+    if (!plugins_list.empty()) {
+        for (int i = 0; i < plugins_list.size(); i++) {
+            plugins_list[i]->make_visible(p_visible);
+        }
+    }
+}
+
+void EditorPluginList::edit(Object* p_object) {
+    if (!plugins_list.empty()) {
+        for (int i = 0; i < plugins_list.size(); i++) {
+            plugins_list[i]->edit(p_object);
+        }
+    }
+}
+
+bool EditorPluginList::forward_input_event(const InputEvent& p_event) {
+    bool discard = false;
+    if (!plugins_list.empty()) {
+        for (int i = 0; i < plugins_list.size(); i++) {
+            if (plugins_list[i]->forward_input_event(p_event)) {
+                discard = true;
```

Comment:
IMO we should exit the loop if the event is captured, avoiding the rest of the plugins to receive it.